pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -592,53 +585,37 @@ func (s *DiscoveryServer) pushConnection(con *XdsConnection, pushEv *XdsEvent) e
 
 	adsLog.Infof("Pushing %v", con.ConID)
 
-	_ = s.rateLimiter.Wait(context.TODO()) // rate limit the actual push
-
 	// Prevent 2 overlapping pushes.
 	con.pushMutex.Lock()
 	defer con.pushMutex.Unlock()
 
 	defer func() {
```

Comment:
Is pushmutex lock still needed? I thought we will never call pushConnection multiple times.